stl/inc/atomic

```diff
@@ -133,6 +117,37 @@ _NODISCARD extern "C" bool __cdecl __std_atomic_has_cmpxchg16b() noexcept;
 
 _STD_BEGIN
 
+// FENCES
+extern "C" inline void atomic_thread_fence(const memory_order _Order) noexcept {
+    if (_Order == memory_order_relaxed) {
+        return;
+    }
+
+#if defined(_M_ARM) || defined(_M_ARM64)
+    _Memory_barrier();
+#else // ^^^ ARM32/ARM64 hardware / x86/x64 hardware vvv

```

Comment:
Is the implementation for `x86/x64` likely safe on any other hypothetical architecture? Assumptions about architecture is what has gotten us into this mess of mediocre ARM support we're in now.

Reply:
Good question.  It is both unsafe, and possibly inefficient where it is safe.

Reply:
So I've addressed this place.

But there are some places like here:
https://github.com/microsoft/STL/blob/4deaf6cb27716170e251ed89fd0d710235ea6995/stl/inc/atomic#L489-L495

I think any of the implementations looks working (apart from `_Memory_barrier()` undefined on x86/x64, but it can be defined). But it still does not make sense to use the wrong implementation for each platform for performance reasons. Should compilation be broken for unknown platform here as well?

Reply:
I'm more okay with it in this case. `_InterlockedExchange16` will likely be defined on future architectures. Also you're not directly touching this so no reason to hold up the PR on that.